In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import re
from datetime import datetime

In [2]:
# ------------------------------------------------------------------------------------------------------------
## Vamos a cargar los datos de Investigadores UTB...
# ------------------------------------------------------------------------------------------------------------
datautb=pd.read_excel("Base de Datos Scopus.xlsx")
datautb=datautb.rename(columns={"ID SCOPUS":"Autor ID"})
datautb["Autor ID"]=datautb["Autor ID"].astype("str")

In [3]:
# Generar query para Scopus: https://www.scopus.com/search/form.uri?display=advanced
search1,search2=[],[]
for j,i in enumerate(datautb["Autor ID"]):
    print("AU-ID ( {} )".format(i)+" OR ", end=' ')

AU-ID ( 57156565000 ) OR  AU-ID ( 14622047600 ) OR  AU-ID ( 24329839300 ) OR  AU-ID ( 57194034904 ) OR  AU-ID ( 57193533853 ) OR  AU-ID ( 24537991200 ) OR  AU-ID ( 36142156300 ) OR  AU-ID ( 57039103600 ) OR  AU-ID ( 56674579200 ) OR  AU-ID ( 57210822856 ) OR  AU-ID ( 57191333650 ) OR  AU-ID ( 7005142049 ) OR  AU-ID ( 57193012270 ) OR  AU-ID ( 55649334800 ) OR  AU-ID ( 26325154200 ) OR  AU-ID ( 55872162200 ) OR  AU-ID ( 35788581800 ) OR  AU-ID ( 57024211000 ) OR  AU-ID ( 56801043600 ) OR  AU-ID ( 56380539800 ) OR  AU-ID ( 57188841051 ) OR  AU-ID ( 57202285682 ) OR  AU-ID ( 57350116000 ) OR  AU-ID ( 58525252300 ) OR  AU-ID ( 57192930752 ) OR  AU-ID ( 57200615582 ) OR  AU-ID ( 58660078000 ) OR  AU-ID ( 57190688459 ) OR  AU-ID ( 57208719994 ) OR  AU-ID ( 55258973100 ) OR  AU-ID ( 57203321995 ) OR  AU-ID ( 57220927199 ) OR  AU-ID ( 57193252278 ) OR  AU-ID ( 57322375300 ) OR  AU-ID ( 57205400052 ) OR  AU-ID ( 57758796500 ) OR  AU-ID ( 56581610900 ) OR  AU-ID ( 57392556500 ) OR  AU-ID ( 56682

### Copiar y pegar 
el último statement y agregar a la linea siguiente. Esta será usada en el sitio de "Advance Search" de Scopus: https://www.scopus.com/search/form.uri?display=advanced...

(AF-ID ( 60103889 )) OR ( AU-ID ( 57156565000 ) OR  AU-ID ( 14622047600 ) OR  AU-ID ( 24329839300 ) OR  AU-ID ( 57194034904 ) OR  AU-ID ( 57193533853 ) OR  AU-ID ( 24537991200 ) OR  AU-ID ( 36142156300 ) OR  AU-ID ( 57039103600 ) OR  AU-ID ( 56674579200 ) OR  AU-ID ( 57210822856 ) OR  AU-ID ( 57191333650 ) OR  AU-ID ( 7005142049 ) OR  AU-ID ( 57193012270 ) OR  AU-ID ( 55649334800 ) OR  AU-ID ( 26325154200 ) OR  AU-ID ( 55872162200 ) OR  AU-ID ( 35788581800 ) OR  AU-ID ( 57024211000 ) OR  AU-ID ( 56801043600 ) OR  AU-ID ( 56380539800 ) OR  AU-ID ( 57188841051 ) OR  AU-ID ( 57202285682 ) OR  AU-ID ( 57350116000 ) OR  AU-ID ( 58525252300 ) OR  AU-ID ( 57192930752 ) OR  AU-ID ( 57200615582 ) OR  AU-ID ( 58660078000 ) OR  AU-ID ( 57190688459 ) OR  AU-ID ( 57208719994 ) OR  AU-ID ( 55258973100 ) OR  AU-ID ( 57203321995 ) OR  AU-ID ( 57220927199 ) OR  AU-ID ( 57193252278 ) OR  AU-ID ( 57322375300 ) OR  AU-ID ( 57205400052 ) OR  AU-ID ( 57758796500 ) OR  AU-ID ( 56581610900 ) OR  AU-ID ( 57392556500 ) OR  AU-ID ( 56682785300 ) OR  AU-ID ( 57219403758 ) OR  AU-ID ( 57197327858 ) OR  AU-ID ( 57196040759 ) OR  AU-ID ( 57220077867 ) OR  AU-ID ( 57750422100 ) OR  AU-ID ( 57223851529 ) OR  AU-ID ( 57189892062 ) OR  AU-ID ( 57205658483 ) OR  AU-ID ( 58068307100 ) OR  AU-ID ( 57197807415 ) OR  AU-ID ( 57918628600 ) OR  AU-ID ( 57222278899 ) OR  AU-ID ( 7195913974 ) OR  AU-ID ( 57218294431 ) OR  AU-ID ( 57206773929 ) OR  AU-ID ( 58068069000 ) OR  AU-ID ( 57216868622 ) OR  AU-ID ( 57930663300 ) OR  AU-ID ( 58523557700 ) OR  AU-ID ( 57204842254 ) OR  AU-ID ( 58618811100 ) OR  AU-ID ( 57201036449 ) OR  AU-ID ( 57209248085 ) OR  AU-ID ( 57212006168 ) OR  AU-ID ( 57427876200 ) OR  AU-ID ( 57204847841 ) OR  AU-ID ( 57195913859 ) OR  AU-ID ( 57218297655 ) OR  AU-ID ( 57222223605 ) OR  AU-ID ( 57221229836 ) OR  AU-ID ( 58803522300 ) OR  AU-ID ( 55783129400 ) OR  AU-ID ( 57903699900 ) OR  AU-ID ( 58954857900 ) OR  AU-ID ( 57202159706 ) OR  AU-ID ( 57219626251 ) OR  AU-ID ( 57215557867 ) OR  AU-ID ( 57219506381 ) OR  AU-ID ( 58134344400 ) OR  AU-ID ( 58917134900 ) OR  AU-ID ( 58153979500 ) OR  AU-ID ( 58868256900 ) OR  AU-ID ( 57918232800 ) OR  AU-ID ( 59987226300 ) )

### Copiar y pegar 
esta linea anterior en: https://www.scopus.com/search/form.uri?display=advanced, descargar el archivo que genera con los docmentos.

In [4]:
## ------------------------------------------------------------------------------------------------------------------------
## Cargando el archivo de datos de publicaciones Scopus descargado desde la página de Scopus
## ------------------------------------------------------------------------------------------------------------------------
scopus=pd.read_csv("scopus-2.csv")

In [5]:
scopus

Authors  \
0                 Herrera-Acevedo D.D.; Sierra-Porta D.   
1                          Geney E.R.; Vázquez-Miraz P.   
2     Medina-Reyes M.F.; Fajardo-Cuadro J.G.; Martín...   
3         Pérez-Morón J.M.; García Alonso R.; Thoene U.   
4     Hernandez-Fernandez J.; Herrera Zabala J.E.; M...   
...                                                 ...   
1683             Ortiz O.O.; Pasqualino J.; Castells F.   
1684           Romero L.A.; Millán M.S.; Pérez-Cabré E.   
1685  Villa J.L.; Duque M.; Gauthier A.; Rakoto-Rava...   
1686                          Marrugo A.G.; Millán M.S.   
1687            Pasqualino J.C.; Montané D.; Salvadó J.   

                                      Author full names  \
0     Herrera-Acevedo, D.D. (59278790000); Sierra-Po...   
1     Geney, Elías Rafael (57427876200); Vázquez-Mir...   
2     Medina-Reyes, María Fernanda (57221229836); Fa...   
3     Pérez-Morón, James Manuel (57322375300); Garcí...   
4     Hernandez-Fernandez, Joaquin (57210432772); He...   
...                                                 ...   
1683  Ortiz, O.O. (24436854200); Pasqualino, J. (146...   
1684  Romero, Lenny A. (36142156300); Millán, María ...   
1685  Villa, J.L. (7005142049); Duque, M. (700486442...   
1686  Marrugo, Andrés G. (24329839300); Millán, Marí...   
1687  Pasqualino, Jorgelina C. (14622047600); Montan...   

                                        Author(s) ID  \
0                           59278790000; 57191333650   
1                           57427876200; 57196040759   
2              57221229836; 56581610900; 26325154200   
3              57322375300; 57215572817; 56766193900   
4              57210432772; 59387631800; 36800050600   
...                                              ...   
1683            24436854200; 14622047600; 7003612213   
1684             36142156300; 7201466399; 8729551400   
1685  7005142049; 7004864427; 7102084702; 6603389671   
1686                         24329839300; 7201466399   
1687            14622047600; 55990834800; 7004090328   

                                                  Title  Year  \
0     Network structure and urban mobility sustainab...  2025   
1     Interpersonal values associated with unfaithfu...  2025   
2     DRIVING THE DEVELOPMENT OF ENERGY COMMUNITIES ...  2025   
3     Looking back to move forward: shedding light o...  2024   
4     Applied Investigation of Methyl, Ethyl, Propyl...  2024   
...                                                 ...   ...   
1683  Environmental assessment based on life cycle a...  2010   
1684  Optical implementation of multifocal programma...  2011   
1685  Supervision and optimal control of a class of ...  2003   
1686  Optic disc segmentation in retinal images; [Se...  2010   
1687  Synergic effects of biodiesel in the biodegrad...  2006   

                                           Source title Volume    Issue  \
0                        Sustainable Cities and Society    119      NaN   
1                          Revista de Psicologia (Peru)     43        1   
2                                   Habitat Sustentable     15        1   
3               New England Journal of Entrepreneurship     27        2   
4                                              Polymers     16       20   
...                                                 ...    ...      ...   
1683                                           Afinidad     67      547   
1684              Journal of Physics: Conference Series    274        1   
1685  IEEE International Conference on Emerging Tech...      2  January   
1686                             Optica Pura y Aplicada     43        2   
1687                              Biomass and Bioenergy     30       10   

     Art. No. Page start Page end  Page count  Cited by  \
0      106076        NaN      NaN         NaN         3   
1         NaN        438      459        21.0         0   
2         NaN         10       19         9.0         0   
3         NaN        152      172      

In [6]:
## ------------------------------------------------------------------------------------------------------------------------
## Cargando el archivo de datos de publicaciones Scimago descargado desde la página de Scimago
## ------------------------------------------------------------------------------------------------------------------------
#scimago=pd.read_excel("scimagojr2024.xlsx")
scimago=pd.read_csv("scimagojr2024.csv",delimiter=";")

In [7]:
scimago

Rank     Sourceid                                              Title  \
0          1        28773                 Ca-A Cancer Journal for Clinicians   
1          2        19434                   MMWR Recommendations and Reports   
2          3        20315              Nature Reviews Molecular Cell Biology   
3          4        29431                     Quarterly Journal of Economics   
4          5        20425                      Nature Reviews Drug Discovery   
...      ...          ...                                                ...   
31131  31132       145515   Waves in Random and Complex Media (discontinued)   
31132  31133  21101174249                             Word and Music Studies   
31133  31134        18033            World Dredging, Mining and Construction   
31134  31135  21100898632  World Scientific-Now Publishers Series in Busi...   
31135  31136  18200156701  Zeitschrift Kunststofftechnik/Journal of Plast...   

                Type                Issn      SJR SJR Best Quartile  H index  \
0            journal  15424863, 00079235  145,004                Q1      223   
1            journal  10575987, 15458601   41,754                Q1      155   
2            journal  14710072, 14710080   37,353                Q1      531   
3            journal  00335533, 15314650   35,995                Q1      322   
4            journal  14741784, 14741776   30,506                Q1      412   
...              ...                 ...      ...               ...      ...   
31131        journal  17455049, 17455030      NaN                 -       54   
31132    book series            15660958      NaN                 -        8   
31133  trade journal            10450343      NaN                 -        3   
31134    book series            22513442      NaN                 -        3   
31135        journal            18642217      NaN                 -       10   

       Total Docs. (2024)  Total Docs. (3years)  ...  Ref. / Doc.  %Female  \
0                      43                   122  ...        62,88        -   
1                       6                    15  ...       275,33        -   
2                     125                   337  ...        92,45        -   
3                      48                   143  ...        69,79        -   
4                     247                   718  ...        35,66        -   
...                   ...                   ...  ...          ...      ...   
31131                 308                  1068  ...        49,05        -   
31132                  16                    21  ...        79,44        -   
31133                   2                    32  ...         0,00        -   
31134                   9                    22  ...        31,33        -   
31135                  20                     0  ...        18,20        -   

       Overton SDG         Country            Region  \
0            4  37   United States  Northern America   
1            1   5   United States  Northern America   
2            0  15  United Kingdom    Western Europe   
3           35  27  United Kingdom    Western Europe   
4            1  58  United Kingdom    Western Europe   
...        ...  ..             ...               ...   
31131        0  20  United Kingdom    Western Europe   
31132        0   0     Netherlands    Western Europe   
31133        0   1   United States  Northern America   
31134        0   0       Singapore    Asiatic Region   
31135        0   9         Germany    Western Europe   

                                              Publisher  \
0                               John Wiley and Sons Inc   
1      Centers for Disease Control and Prevention (CDC)   
2                                       Nature Research   
3                               Oxford University Press   
4                                       Nature Research   
...                                                 ...   
31131                           Taylor and Francis Ltd.   
3113

In [8]:
## ------------------------------------------------------------------------------------------------------------------------
## Este pedazo de script cruza ambos archivos anteriores (Scopus y Scimago) para añadir una columna en "Scopus"
## que se llama "Best Quartile" que corresponde a la publicación de cada una de las lineas de "Scopus"...
## ------------------------------------------------------------------------------------------------------------------------

best_quartile = []
for i in range(len(scopus)):
    journal = str(scopus.loc[i, "Source title"]).strip().lower()
    best = scimago.loc[scimago["Title"].str.strip().str.lower() == journal, "SJR Best Quartile"].values
    best_quartile.append(best[0] if len(best) == 1 else np.nan)

scopus["Best Quartile"] = best_quartile

#best_quartile=[]
#for i in range(len(scopus)):
#    journal=scopus["Source title"][i]
#    best=scimago.loc[scimago["Title"]==journal]["SJR Best Quartile"].values
#    if len(best)==1:
#        best_quartile.append(best[0])
#    else:
#        best_quartile.append(np.nan)
#    #print(journal,best)

#scopus["Best Quartile"]=best_quartile
scopus.head(1)

Authors  \
0  Herrera-Acevedo D.D.; Sierra-Porta D.   

                                   Author full names  \
0  Herrera-Acevedo, D.D. (59278790000); Sierra-Po...   

               Author(s) ID  \
0  59278790000; 57191333650   

                                               Title  Year  \
0  Network structure and urban mobility sustainab...  2025   

                     Source title Volume Issue Art. No. Page start  ...  \
0  Sustainable Cities and Society    119   NaN   106076        NaN  ...   

  Page count  Cited by                        DOI  \
0        NaN         3  10.1016/j.scs.2024.106076   

                                                Link Document Type  \
0  https://www.scopus.com/inward/record.uri?eid=2...       Article   

  Publication Stage Open Access  Source                 EID Best Quartile  
0             Final         NaN  Scopus  2-s2.0-85213079490            Q1  

[1 rows x 21 columns]

In [9]:
## ------------------------------------------------------------------------------------------------------------------------
## Este bloque de script genera un nuevo dataframe limpio que contiene todas las publicaciones con información de 
## tipo de publicación, curatil, etc....
## ------------------------------------------------------------------------------------------------------------------------

data_clean = scopus.copy()  # ← tabla final con cuartil
# --------------------------------------------------------------------
# 4. Explosión a nivel autor-artículo
#    (autor ID es muy fiable; si no está, usa nombre)
# --------------------------------------------------------------------
def explode_authors(row):
    ids   = str(row["Author(s) ID"]).split(";")
    names = str(row["Author full names"]).split(";")
    for aid, aname in zip(ids, names):
        aid = aid.strip()
        aname = aname.strip()
        if aid:
            yield {
                "author_id": aid,
                "author_name": aname,
                "year": int(row["Year"]),
                "citations": row.get("Cited by", np.nan),
                "publication": row.get("Title"),
                "journal": row.get("Source title"),
                "volume": row.get("Volume", np.nan),
                "issue": row.get("Issue", np.nan),
                "doi": row.get("DOI", np.nan),
                "document type": row.get("Document Type", np.nan),
                "quartile": row.get("Best Quartile", np.nan)
            }

author_rows = []
for _, r in data_clean.iterrows():
    author_rows.extend(list(explode_authors(r)))

auth_df = pd.DataFrame(author_rows)

In [10]:
auth_df

author_id                                 author_name  year  \
0     59278790000         Herrera-Acevedo, D.D. (59278790000)  2025   
1     57191333650              Sierra-Porta, D. (57191333650)  2025   
2     57427876200           Geney, Elías Rafael (57427876200)  2025   
3     57196040759          Vázquez-Miraz, Pedro (57196040759)  2025   
4     57221229836  Medina-Reyes, María Fernanda (57221229836)  2025   
...           ...                                         ...   ...   
7315  24329839300            Marrugo, Andrés G. (24329839300)  2010   
7316   7201466399               Millán, María S. (7201466399)  2010   
7317  14622047600      Pasqualino, Jorgelina C. (14622047600)  2006   
7318  55990834800               Montané, Daniel (55990834800)  2006   
7319   7004090328                  Salvadó, Joan (7004090328)  2006   

      citations                                        publication  \
0             3  Network structure and urban mobility sustainab...   
1             3  Network structure and urban mobility sustainab...   
2             0  Interpersonal values associated with unfaithfu...   
3             0  Interpersonal values associated with unfaithfu...   
4             0  DRIVING THE DEVELOPMENT OF ENERGY COMMUNITIES ...   
...         ...                                                ...   
7315          4  Optic disc segmentation in retinal images; [Se...   
7316          4  Optic disc segmentation in retinal images; [Se...   
7317        155  Synergic effects of biodiesel in the biodegrad...   
7318        155  Synergic effects of biodiesel in the biodegrad...   
7319        155  Synergic effects of biodiesel in the biodegrad...   

                             journal volume issue  \
0     Sustainable Cities and Society    119   NaN   
1     Sustainable Cities and Society    119   NaN   
2       Revista de Psicologia (Peru)     43     1   
3       Revista de Psicologia (Peru)     43     1   
4                Habitat Sustentable     15     1   
...                              ...    ...   ...   
7315          Optica Pura y Aplicada     43     2   
7316          Optica Pura y Aplicada     43     2   
7317           Biomass and Bioenergy     30    10   
7318           Biomass and Bioenergy     30    10   
7319           Biomass and Bioenergy     30    10   

                                  doi document type quartile  
0           10.1016/j.scs.2024.106076       Article       Q1  
1           10.1016/j.scs.2024.106076       Article       Q1  
2           10.18800/psico.202501.015       Article       Q3  
3           10.18800/psico.202501.015       Article       Q3  
4     10.22320/07190700.2025.15.01.01       Article       Q3  
...                               ...           ...      ...  
7315                              NaN       Article       Q4  
7316                              NaN       Article       Q4  
7317   10.1016/j.biombioe.2006.03.002       Article       Q1  
7318   10.1016/j.biombioe.2006.03.002       Article       Q1  
7319   10.1016/j.biombioe.2006.03.002       Article       Q1  

[7320 rows x 11 columns]

In [11]:
## ------------------------------------------------------------------------------------------------------------------------
## GET h-index-all and h5yr-index
## ------------------------------------------------------------------------------------------------------------------------

def h_index_from_citations(cites):
    """
    cites: iterable de enteros (citas por artículo)
    retorna: h-index (int)
    """
    # Limpieza básica
    vals = [max(0, int(x)) for x in cites if x == x]  # quita NaN y negativos
    vals.sort(reverse=True)
    # cuenta cuántos cumplen c >= i (con i=1..n) en orden descendente
    return sum(c >= i+1 for i, c in enumerate(vals))
    
def h_index_author(auth_df, author_id, min_year=None, max_year=None):
    sub = auth_df[auth_df["author_id"] == str(author_id)]
    if min_year is not None:
        sub = sub[sub["year"] >= min_year]
    if max_year is not None:
        sub = sub[sub["year"] <= max_year]
    return h_index_from_citations(sub["citations"])

In [12]:
# ------------------------------------------------------------------------------------------------------------
# Probando generación de h-index y h5yr-index
# ------------------------------------------------------------------------------------------------------------

# h-index total del autor
h_all = h_index_author(auth_df, "24329839300")
# h-index de los últimos 5 años (ajusta el rango que quieras)
import datetime
y = datetime.datetime.now().year
h_5yr = h_index_author(auth_df, "24329839300", min_year=y-5, max_year=y)
print(h_all, h_5yr)

13 10


In [ ]:
q1_2025,q2_2025,q3_2025,q4_2025,sq_2025=[],[],[],[],[]
q1_2024,q2_2024,q3_2024,q4_2024,sq_2024=[],[],[],[],[]
q1_2023,q2_2023,q3_2023,q4_2023,sq_2023=[],[],[],[],[]
q1_2022,q2_2022,q3_2022,q4_2022,sq_2022=[],[],[],[],[]
h,h5,autor_sc,nameautor=[],[],[],[]
y = datetime.datetime.now().year
for i in auth_df["author_id"].unique():
    autor_sc.append(i)
    nameautor.append(auth_df[auth_df["author_id"]==i]["author_name"].unique()[0])
    h.append(h_index_author(auth_df, i))
    h5.append(h_index_author(auth_df, i, min_year=y-5, max_year=y))
    q1_2025.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2025)&(auth_df["quartile"]=="Q1")]))
    q2_2025.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2025)&(auth_df["quartile"]=="Q2")]))
    q3_2025.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2025)&(auth_df["quartile"]=="Q3")]))
    q4_2025.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2025)&(auth_df["quartile"]=="Q4")]))
    sq_2025.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2025)&\
        ((auth_df["quartile"]!="Q1")&(auth_df["quartile"]!="Q2")&(auth_df["quartile"]!="Q3")&(auth_df["quartile"]!="Q4"))]))
    q1_2024.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2024)&(auth_df["quartile"]=="Q1")]))
    q2_2024.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2024)&(auth_df["quartile"]=="Q2")]))
    q3_2024.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2024)&(auth_df["quartile"]=="Q3")]))
    q4_2024.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2024)&(auth_df["quartile"]=="Q4")]))
    sq_2024.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2024)&\
        ((auth_df["quartile"]!="Q1")&(auth_df["quartile"]!="Q2")&(auth_df["quartile"]!="Q3")&(auth_df["quartile"]!="Q4"))]))
    q1_2023.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2023)&(auth_df["quartile"]=="Q1")]))
    q2_2023.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2023)&(auth_df["quartile"]=="Q2")]))
    q3_2023.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2023)&(auth_df["quartile"]=="Q3")]))
    q4_2023.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2023)&(auth_df["quartile"]=="Q4")]))
    sq_2023.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2023)&\
        ((auth_df["quartile"]!="Q1")&(auth_df["quartile"]!="Q2")&(auth_df["quartile"]!="Q3")&(auth_df["quartile"]!="Q4"))]))
    q1_2022.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2022)&(auth_df["quartile"]=="Q1")]))
    q2_2022.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2022)&(auth_df["quartile"]=="Q2")]))
    q3_2022.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2022)&(auth_df["quartile"]=="Q3")]))
    q4_2022.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2022)&(auth_df["quartile"]=="Q4")]))
    sq_2022.append(len(auth_df[(auth_df["author_id"]==i)&(auth_df["year"]==2022)&\
        ((auth_df["quartile"]!="Q1")&(auth_df["quartile"]!="Q2")&(auth_df["quartile"]!="Q3")&(auth_df["quartile"]!="Q4"))]))

In [ ]:
import datetime as dt
year_now=pd.Timestamp.now().year
summary_final=pd.DataFrame({"Autor ID":autor_sc,"Autor Name":nameautor,
                            "H-index":h,"H5yr-index":h5,
                            "Q1_2025":q1_2025,"Q2_2025":q2_2025,"Q3_2025":q3_2025,"Q4_2025":q4_2025,"SQ_2025":sq_2025,
                            "Q1_2024":q1_2024,"Q2_2024":q2_2024,"Q3_2024":q3_2024,"Q4_2024":q4_2024,"SQ_2024":sq_2024,
                            "Q1_2023":q1_2023,"Q2_2023":q2_2023,"Q3_2023":q3_2023,"Q4_2023":q4_2023,"SQ_2023":sq_2023,
                            "Q1_2022":q1_2022,"Q2_2022":q2_2022,"Q3_2022":q3_2022,"Q4_2022":q4_2022,"SQ_2022":sq_2022})

In [ ]:
summary_final.sort_values(by="H-index",ascending=False).head(2)

In [ ]:
datautb.set_index("Autor ID",inplace=True)
summary_final.set_index("Autor ID",inplace=True)
summary_final["Total 2025"]=summary_final[["Q1_2025","Q2_2025","Q3_2025","Q4_2025"]].sum(axis=1)
summary_final["Total 2025 (include SQ)"]=summary_final[["Q1_2025","Q2_2025","Q3_2025","Q4_2025","SQ_2025"]].sum(axis=1)
summary_final["Total 2024"]=summary_final[["Q1_2024","Q2_2024","Q3_2024","Q4_2024"]].sum(axis=1)
summary_final["Total 2024 (include SQ)"]=summary_final[["Q1_2024","Q2_2024","Q3_2024","Q4_2024","SQ_2024"]].sum(axis=1)
summary_final["Total 2023"]=summary_final[["Q1_2023","Q2_2023","Q3_2023","Q4_2023"]].sum(axis=1)
summary_final["Total 2023 (include SQ)"]=summary_final[["Q1_2023","Q2_2023","Q3_2023","Q4_2023","SQ_2023"]].sum(axis=1)
summary_final["Total 2022"]=summary_final[["Q1_2022","Q2_2022","Q3_2022","Q4_2022"]].sum(axis=1)
summary_final["Total 2022 (include SQ)"]=summary_final[["Q1_2022","Q2_2022","Q3_2022","Q4_2022","SQ_2022"]].sum(axis=1)

In [ ]:
data1=datautb.merge(summary_final,left_index=True, right_index=True)\
    .sort_values(by=["H-index"],ascending=False)
data1.to_excel("MidataMiScript.xlsx")
data1

In [ ]:
auth_df=auth_df.rename(columns={"author_id":"Autor ID"})
auth_df.set_index("Autor ID",inplace=True)

In [ ]:
data2=datautb.merge(auth_df,left_index=True, right_index=True)\
    .sort_values(by=["year"],ascending=False)
data2.to_excel("MidataMiScript2.xlsx")
data2

In [ ]:
data1.sort_values(by="Total 2025", ascending=False).head(15)

In [ ]:
print("DOne...! Archivos generados...!")

Plots...

In [ ]:
dat_ETD=data1[(data1["ESCUELA"]=="ESCUELA DE TRANSFORMACIÓN DIGITAL")&(data1["Total 2025 (include SQ)"]>0)]

In [ ]:
dat_ETD.set_index("DOCENTE")\
    .sort_values(by="Total 2025",ascending=True)[["Q1_2025","Q2_2025","Q3_2025","Q4_2025","SQ_2025"]]\
    .plot.barh(stacked=True, figsize=(12, 4))
plt.grid(alpha=0.3)
plt.savefig("ETD1.png",dpi=300,bbox_inches='tight')

In [ ]:
dat_ETD[["Q1_2025","Q2_2025","Q3_2025","Q4_2025","SQ_2025"]].T.sum(axis=1).plot.pie(autopct='%1.1f%%')
plt.savefig("ETD2.png",dpi=300,bbox_inches='tight')
plt.show()